design/hardware-classification-controller/support-for-new-parameters-hwcc.md

```diff
@@ -0,0 +1,243 @@
+# Add Support for extra hardware parameters in Disk, RAM, CPU and NIC
+ 
+## Table of Contents
+ 
+<!--ts-->
+   * [Hardware Classification](#title)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+      * [Proposal](#proposal)
+         * [Investigation Details](#investigation-details)
+         * [Implementation Details/Notes/Constraints ](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigation](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+      * [References](#references)
+
+## Summary
+ 
+Implement support for more parameters in HWCC which will help to 
+classify a baremetal host on more possible configurations. 
+ 
+## Motivation
+
+We have hardware classification controller which validates and 
+classifies the hosts based on user hardware requirements.
+We would like to add more parameters which helps user to 
+identify the ideal host. This feature helps user to avoid runtime failures
+and also increases the performance for workload deployments.
+
+### Goals
+
+Automatically classify bare metal hosts based on user-provided rules
+in a way that other controllers can use, without having to evaluate
+all of those rules themselves.
```

Comment:
This is really the goal of the whole hardware classification controller project, right? Is there a more narrowly defined goal that caused you to choose the specific fields being proposed here?

Reply:
Yes, it looks like that. We will update it and make it more specific to this Proposal.